common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerRequest.java

```diff
@@ -155,6 +157,18 @@ private static final class SerializedNames {
     @SerializedName(SerializedNames.PROMPT)
     private String mPrompt;
 
+    /**
+     * The child redirect uri for the request.
+     */
+    @SerializedName(SerializedNames.CHILD_REDIRECT_URI)
+    private String mChildRedirectUri;
```

Comment:
I thought the child one was being called as the normal client id/ redirect uri and the parent one gets the name of broker client id/redirect uri

Reply:
from OneAuth's perspective or client's perspective, they send 
child parameters as childClientId and chileRedirectURI  and hub app's parameters as clientId and redirectURI.

We read these params in broker
1. While doing a silent call, we would not have to make any changes to do a cache lookup -- the lookup still happens with clientId and redirectURI to get RT (hub app's RT)
2. While doing an interactive call, we would have to send clientId,  redirectURI as child params and brkClientId, brkRedirectURI as hub parameters. 

So, in the changes I made, you would see only TokenRequest and JwtRequestBody would be having new brkClientId, brkRedirect parameters added.

If we instead change to the way you are suggesting - to only use childClientId and childRedirectURi and use serialzation to change the names, we would have to change TokenRequest to below

  @SerializedName(SerializedNames.REDIRECT_URI)
    private String mChildRedirectUri;

  @SerializedName(SerializedNames.BRK_REDIRECT_URI)
    private String mRedirectUri;

This is even more confusing.. and this would mean, even in existing requests (without NAA), wherever we are setting mRedirectURI, we would be sending brk_redirect_uri to the service.

Reply:
Could we add a new method callled `createNaaTokenRequest` that can achieve this?